Python/gc_free_threading.c

```diff
@@ -433,6 +433,12 @@ static void
 gc_visit_thread_stacks(PyInterpreterState *interp, struct collection_state *state)
 {
     _Py_FOR_EACH_TSTATE_BEGIN(interp, p) {
+        _PyCStackRef *c_ref = ((_PyThreadStateImpl *)p)->c_stack_refs;
+        while (c_ref != NULL) {
+            gc_visit_stackref(c_ref->ref);
```

Comment:
This call seems unsafe if the `_PyCStackRef` is not initialized before `gc_visit_stackref()` is called on it. `_PyCStackRef cref;` creates uninitialized memory (on the stack).

Does your design rely on the fact that the `_PyCStackRef` is initialized before a GC collection can be triggered?

I suggest to either document it, or always initialize `_PyCStackRef.ref`.

Reply:
Oh, I missed the last line of `_PyThreadState_PushCStackRef()`: `ref->ref = PyStackRef_NULL;`.

In this case, I changed my mind: the code is safe ;-)